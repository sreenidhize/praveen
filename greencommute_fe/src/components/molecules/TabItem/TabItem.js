import { makeStyles, Typography } from "@material-ui/core";
import React from "react";
import PropTypes from "prop-types";

const useStyle = makeStyles((theme) => ({
  tabActive: {
    width: "240px",
    height: "50px",
    borderRadius: "10px",
    backgroundColor: "rgba(90, 197, 104, 0.15)",
    display: "flex",
    cursor: "pointer",
  },
  tabInActive: {
    width: "240px",
    height: "50px",
    display: "flex",
    cursor: "pointer",
  },
  inActiveTextClass: {
    ...theme.typography.navtabs,
    paddingLeft: "10%",
    alignSelf: "center",
    color: theme.typography.h4.color,
    whiteSpace: "nowrap",
    overflow: "hidden",
    textOverflow: "ellipsis",
  },
  activeTextClass: {
    ...theme.typography.navtabs,
    paddingLeft: "10%",
    alignSelf: "center",
    color: theme.palette.primary.main,
    whiteSpace: "nowrap",
    overflow: "hidden",
    textOverflow: "ellipsis",
  },
  inActiveIcon: {
    paddingLeft: "8%",
    alignSelf: "center",
    color: theme.typography.h4.color,
  },
  activeIcon: {
    paddingLeft: "8%",
    alignSelf: "center",
    color: theme.palette.primary.main,
  },
}));

function TabItem({ isActive, Icon, label, tabOnClick, dataTestid }) {
  const classes = useStyle();
  let divClass = isActive ? classes.tabActive : classes.tabInActive;

  return (
    <div className={divClass} onClick={tabOnClick} data-testid={dataTestid}>
      {isActive && (
        <>
          <Icon className={classes.activeIcon} />
          <Typography className={classes.activeTextClass}>{label}</Typography>
        </>
      )}
      {!isActive && (
        <>
          <Icon className={classes.inActiveIcon} />
          <Typography className={classes.inActiveTextClass}>{label}</Typography>
        </>
      )}
    </div>
  );
}

TabItem.propTypes = {
  isActive: PropTypes.bool,
  label: PropTypes.string,
  Icon: PropTypes.object,
  tabOnClick: PropTypes.func,
  dataTestid: PropTypes.string,
};

TabItem.defaultProps = {};

export default TabItem;
