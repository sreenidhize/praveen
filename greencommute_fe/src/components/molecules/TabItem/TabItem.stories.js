import React from "react";
import TabItem from "./TabItem";
import WorkOutlineIcon from "@material-ui/icons/WorkOutline";

const TabItemStory = {
  title: "Molecules/TabItem",
  component: TabItem,
};

const Template = (args) => <TabItem {...args} />;

export const Primary = Template.bind({});

Primary.args = {
  Icon: WorkOutlineIcon,
  label: "Find Jobs",
  isActive: true,
};
export default TabItemStory;
