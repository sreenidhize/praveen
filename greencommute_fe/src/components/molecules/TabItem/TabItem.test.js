import React from "react";
import TabItem from "./TabItem";
import { render, screen } from "@testing-library/react";
import WorkOutlineIcon from "@material-ui/icons/WorkOutline";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";

describe("TabItem renders correctly", () => {
  const onClick = jest.fn();
  test("should match snapshot", () => {
    const { asFragment } = render(
      <TabItem
        Icon={WorkOutlineIcon}
        label="Find Jobs"
        isActive={true}
        tabOnClick={onClick}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("TabItem should have label and icon", () => {
    const { container } = render(
      <TabItem
        Icon={WorkOutlineIcon}
        label="Find Jobs"
        isActive={true}
        tabOnClick={onClick}
        dataTestid="Tab1"
      />
    );
    expect(container.getElementsByTagName("svg").length).toBe(1);
    expect(container.getElementsByTagName("p").length).toBe(1);
    expect(screen.getByText("Find Jobs")).toHaveTextContent("Find Jobs");
    userEvent.click(screen.getByTestId("Tab1"));
    expect(onClick).toHaveBeenCalled();
  });
});
