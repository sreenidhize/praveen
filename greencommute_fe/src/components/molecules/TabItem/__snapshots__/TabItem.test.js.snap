// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabItem renders correctly should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-tabActive-1"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root makeStyles-activeIcon-6"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M14 6V4h-4v2h4zM4 8v11h16V8H4zm16-2c1.11 0 2 .89 2 2v11c0 1.11-.89 2-2 2H4c-1.11 0-2-.89-2-2l.01-11c0-1.11.88-2 1.99-2h4V4c0-1.11.89-2 2-2h4c1.11 0 2 .89 2 2v2h4z"
        fill-rule="evenodd"
      />
    </svg>
    <p
      class="MuiTypography-root makeStyles-activeTextClass-4 MuiTypography-body1"
    >
      Find Jobs
    </p>
  </div>
</DocumentFragment>
`;
